language-server/src/dotty/tools/languageserver/ParsedComment.scala

```diff
@@ -0,0 +1,202 @@
+
+  /** The content of this comment, after expansion if possible. */
+  val content: String = comment.expandedBody.getOrElse(comment.raw)
+
+  /** An index that marks all sections boundaries */
+  private lazy val tagIndex: List[Bounds] = CommentParsing.tagIndex(content)
+
+  /**
+   * Maps a parameter name to the bounds of its doc
+   *
+   * @see paramDoc
+   */
+  private lazy val paramDocs: Map[String, Bounds] = CommentParsing.paramDocs(content, "@param", tagIndex)
+
+  /**
+   * The "main" documentation for this comment. That is, the comment before any section starts.
+   */
+  lazy val mainDoc: String = {
+    val doc = tagIndex match {
+      case Nil => content.stripSuffix("*/")
+      case (start, _) :: _ => content.slice(0, start)
+    }
+    clean(doc.stripPrefix("/**"))
+  }
+
+  /**
+   * Renders this comment as markdown.
+   *
+   * The different sections are formatted according to the maping in `knownTags`.
+   */
+  def renderAsMarkdown: String = {
+    val buf = new StringBuilder
+    buf.append(mainDoc + System.lineSeparator + System.lineSeparator)
+    val groupedSections = CommentParsing.groupedSections(content, tagIndex)
+
+    for {
+      (tag, formatter) <- ParsedComment.knownTags
+      boundss <- groupedSections.get(tag)
+      texts = boundss.map { (start, end) => clean(content.slice(start, end)) }
+      formatted <- formatter(texts)
+    } {
+      buf.append(formatted)
+      buf.append(System.lineSeparator)
+    }
+
+    buf.toString
+  }
+
+  /**
+   * The `@param` section corresponding to `name`.
+   *
+   * @param name The parameter name whose documentation to extract.
+   * @return The formatted documentation corresponding to `name`.
+   */
+  def paramDoc(name: TermName): Option[String] = paramDocs.get(name.toString).map { (start, end) =>
+    val rawContent = content.slice(start, end)
+    val docContent = ParsedComment.prefixRegex.replaceFirstIn(rawContent, "")
+    clean(docContent)
+  }
+
+  /**
+   * Cleans `str`: remove prefixing `*` and trim the string.
+   *
+   * @param str The string to clean
+   * @return The cleaned string.
+   */
+  private def clean(str: String): String = str.stripMargin('*').trim
+}
+
+object ParsedComment {
+
+  /**
+   * Return the `ParsedComment` associated with `symbol`, if it exists.
+   *
+   * @param symbol The symbol for which to retrieve the documentation
+   * @return If it exists, the `ParsedComment` for `symbol`.
+   */
+  def docOf(symbol: Symbol)(implicit ctx: Context): Option[ParsedComment] = {
+    val documentedSymbol = if (symbol.isPrimaryConstructor) symbol.owner else symbol
+    for { docCtx <- ctx.docCtx
+          comment <- docCtx.docstring(documentedSymbol)
+    } yield new ParsedComment(comment)
+  }
+
+  private val prefixRegex = """@param\s+\w+\s+""".r
+
+  /** A mapping from tag name to `TagFormatter` */
+  private val knownTags: ListMap[String, TagFormatter] = ListMap(
+    "@tparam"  -> TagFormatter("Type Parameters", toDescriptionList),
```

Comment:
This seems like something that could be useful when displaying documentation outside of the IDE e.g. in the repl, or even to generate the dottydoc output. Can we move it to compiler/ ?

Reply:
I've moved it to `dotty.tools.dotc.util`